db/database.go

```diff
@@ -1455,6 +1455,7 @@ func (db *Database) Compact(ctx context.Context, skipRunningStateCheck bool, cal
 	for _, c := range db.CollectionByID {
 		// create admin collection interface
 		collection, err := db.GetDatabaseCollectionWithUser(c.ScopeName(), c.Name())
+		ctx := base.CollectionNameCtx(ctx, collection.Name())
```

Comment:
Shouldn't this be getting added more generically to the context in handler.go in the case where it's a keyspace endpoint, and passed through with the ctx parameter?

Reply:
I'm not sure I understand this question. 

Compact may be executed from an endpoint or through a Background task. If accessed through an endpoint it's stamped with a db (since this request is db scoped) and a request id.

I override the context here only inside the loop which becomes collection scoped.

Reply:
Makes sense, and agree with overriding it here given the usage.